cumulusci/cli/cci.py

```diff
@@ -217,26 +218,41 @@ def main(args=None):
         try:
             cli(standalone_mode=False)
         except click.Abort:  # Keyboard interrupt
-            click.echo("\nAborted!")
+            show_debug_info() if debug else click.echo("\nAborted!")
+            sys.exit(1)
+        except exceptions.ConnectionError as e:
+            show_debug_info() if debug else handle_connection_error(e)
             sys.exit(1)
         except Exception as e:
-            # Display the error
-            if debug:
-                traceback.print_exc()
-                pdb.post_mortem()
-            else:
-                click.echo(click.style(f"Error: {e}", fg="red"))
-                # Only suggest gist command if it wasn't run
-                if not is_gist_command:
-                    click.echo(click.style(SUGGEST_GIT_GIST_COMMAND, fg="yellow"))
-
-                with open(CCI_LOGFILE_PATH, "a") as log_file:
-                    traceback.print_exc(file=log_file)  # log stacktrace silently
-            # TODO: errorsdb
-            # Return a non-zero exit code to indicate a problem
+            show_debug_info() if debug else handle_generic_error(e, is_gist_command)
             sys.exit(1)
 
 
+def handle_connection_error(error):
+    click.echo(
+        click.style(
+            f"We encountered an error with your internet connection. Please check your connection and try the last cci command again.\nError: {error}",
```

Comment:
This message could probably be improved. Let me know if anyone has suggestions.

Reply:
Hmm. The connection error could be due to a local network issue, but it could be a problem on the server side that the user doesn't have control over too. In that case it would be nice to have the traceback logged so we can get a clue as to what cci was trying to connect to.

Reply:
@Br4nd0R Sorry, I wasn't very clear about what I meant. I didn't mean that we should output the stack trace to stdout (we just did a bunch of refactoring to avoid doing that except for when the user asks for it). Instead what I was suggesting is that connection errors should not have their own `except` block in `main()`, because we don't want to bypass the normal error handling logic (with its support for the debug flag, logging the traceback to the log file, etc). The only difference for connection errors is that we want to print a custom message instead of the message from the exception instance.

Reply:
Got it.